services/src/util/apidoc.rs

```diff
@@ -148,19 +161,31 @@ impl Modify for TransformSchemasWithTag {
                 let Some(variant_tag) = Self::get_variant_tag(item, discriminator) else {
                     continue 'outer;
                 };
-                let variant_schema_name = match Self::get_base_type_name(item) {
-                    Some(base_type) => format!(
-                        "{}With{}",
-                        base_type,
-                        Self::uppercase_first_letter(discriminator),
-                    ),
-                    None => format!(
-                        "{}{}",
-                        Self::uppercase_first_letter(variant_tag),
-                        schema_name
-                    ),
+
+                let variant_schema_name = match Self::get_title(item) {
+                    // Always prefer custom title as variant name.
+                    Some(title) => title.to_owned(),
+                    None => match Self::get_base_type_name(item) {
```

Comment:
can you add tests for the cases Some(..) and None?

Reply:
The already existing test `separate_inline_schema` goes into both cases.